Source/SwiftLintFramework/Rules/Idiomatic/RandomRule.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+import SourceKittenFramework
+
+public struct RandomRule: OptInRule, ConfigurationProviderRule, AutomaticTestableRule {
+
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static var description = RuleDescription(
+        identifier: "random",
+        name: "Random",
+        description: "Prefer using `type.random(in:)` over C-based functions.",
+        kind: .lint,
+        minSwiftVersion: .fourDotTwo,
+        nonTriggeringExamples: [
+            "Int.random(in: 0..<10)\n",
+            "Double.random(in: 8.6...111.34)\n",
+            "Float.random(in: 0 ..< 1)\n"
+        ],
+        triggeringExamples: [
+            "↓arc4random(10)\n",
+            "↓arc4random_uniform(83)\n",
+            "↓drand48(52)\n"
+        ]
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        let pattern = "(arc4random|drand48)"
```

Comment:
this would be more reliably implemented by using an `ASTRule` checking for function calls. `LegacyConstructorRule` would be an example (although it's probably more complicated than what you need)